### PR TITLE
Issue 105

### PR DESCRIPTION
**Pull request by tapthaker**

### CODE REVIEW

File: _drafts/2018-03-23-issue-105.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+{% include task_remind.html %}
+
+### Swift Unwrapped
+
+[50: Swift 4.1 w/ Doug & Ben (part 1)](https://spec.fm/podcasts/swift-unwrapped/125759): Jessie & JP go over the Swift 4.1 release highlights with Ben Cohen and Doug Gregor from the Swift team.
+
+### News and community
+
+
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: If there is no news to put here, you can remove the section.

File: _drafts/2018-03-23-issue-105.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+### Commits and pull requests
+
+[Karoy Lorentey](https://github.com/lorentey) merged a [pull request](https://github.com/apple/swift/pull/14913) which made Swift use a much higher-quality hashing with a random seeding. You can read about this change in [swift forums](https://forums.swift.org/t/psa-the-stdlib-now-uses-randomly-seeded-hash-values/10789)
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: `with random seeding`

Comment (BasThomas) [reply _drafts/2018-03-23-issue-105.md]: `You can read about this change on the*`

Comment (BasThomas) [reply _drafts/2018-03-23-issue-105.md]: And please capitalize "Swift". :)

Comment (BasThomas) [reply _drafts/2018-03-23-issue-105.md]: ... plus a `.` at the end of the sentence.

File: _drafts/2018-03-23-issue-105.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+>“Random seeding” means that hashValue properties will return different values on each execution of a Swift program. This is an important tool for improving the reliability of the Standard Library’s hashing collections, `Set` and `Dictionary`. In particular, random seeding enables better protection against (accidental or deliberate) [hash-flooding attacks](https://arstechnica.com/information-technology/2011/12/huge-portions-of-web-vulnerable-to-hashing-denial-of-service-attack/).
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: Lets put `hashValue` in backticks.

File: _drafts/2018-03-23-issue-105.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+[Joe Groff](https://github.com/jckarter) merge a [pull request](https://github.com/apple/swift/pull/15145/) that made generic parameter counts in context descriptors 16-bit from 32-bit reducing the metadata. 64K generic parameters ought to be enough for anyone, right? 🧐
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: `merged*`

Comment (BasThomas) [reply _drafts/2018-03-23-issue-105.md]: `from 32-bit, reducing`

Comment (BasThomas) [reply _drafts/2018-03-23-issue-105.md]: Lets also remove the second sentence; I don't think it adds value here.

File: _drafts/2018-03-23-issue-105.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+>“Random seeding” means that hashValue properties will return different values on each execution of a Swift program. This is an important tool for improving the reliability of the Standard Library’s hashing collections, `Set` and `Dictionary`. In particular, random seeding enables better protection against (accidental or deliberate) [hash-flooding attacks](https://arstechnica.com/information-technology/2011/12/huge-portions-of-web-vulnerable-to-hashing-denial-of-service-attack/).
+
+[Joe Groff](https://github.com/jckarter) merge a [pull request](https://github.com/apple/swift/pull/15145/) that made generic parameter counts in context descriptors 16-bit from 32-bit reducing the metadata. 64K generic parameters ought to be enough for anyone, right? 🧐 
+
+### Accepted proposals
+
+[SE-0194](https://github.com/apple/swift-evolution/blob/master/proposals/0194-derived-collection-of-enum-cases.md) *Derived Collection of Enum Cases*  was [accepted](https://forums.swift.org/t/accepted-se-0194-derived-collection-of-enum-cases/10723) with revisions.
+
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: We should add a short description of the reasoning here.

File: _drafts/2018-03-23-issue-105.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+### Returned proposals
+
+> TODO
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: Feel free to remove.

File: _drafts/2018-03-23-issue-105.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+### Rejected proposals
+
+> TODO
+
+### Proposals in review
+
+[SE-0193](https://github.com/apple/swift-evolution/blob/master/proposals/0193-cross-module-inlining-and-specialization.md) Review of *Cross-module inlining and specialization*
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: What about `[SE-0193](https://github.com/apple/swift-evolution/blob/master/proposals/0193-cross-module-inlining-and-specialization.md): *Cross-module inlining and specialization* is [under extended review](https://forums.swift.org/t/se-0193-cross-module-inlining-and-specialization/7310/59).

File: _drafts/2018-03-23-issue-105.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+ [extended](https://forums.swift.org/t/se-0193-cross-module-inlining-and-specialization/7310/59) to resolve naming of 'abiPublic' attribute.
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: Then remove this line, as it is already discussed in the excerpt.

File: _drafts/2018-03-23-issue-105.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+ 
+> This proposal will essentially be accepted as is — except for the name for @abiPublic. Many people voiced concern for this name. While it technically is accurate, it exposes the term “abi” which will be foreign to many. It also doesn’t connote the full implications and value of adding the attribute.
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: Lets backtick `@abiPublic` here.

File: _drafts/2018-03-23-issue-105.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+ 
+[SE-0192](https://github.com/apple/swift-evolution/blob/master/proposals/0192-non-exhaustive-enums.md) *Non-Exhaustive Enums* is in [re-review](https://forums.swift.org/t/se-0192-non-exhaustive-enums/7291).
+ 
+> The second review of the review of SE 0192 — Non-Exhaustive Enums begins now and runs through March 23, 2018.
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: You can omit this line.

File: _drafts/2018-03-23-issue-105.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+>
+>  The second review is follow up to the [original review](https://forums.swift.org/t/se-0192-non-exhaustive-enums/7291) based on feedback on the thread and from the Core Team.
+
+### Swift Forums
+
+[Ankit Aggarwal](https://github.com/aciidb0mb3r) & [Daniel Dunbar](https://github.com/ddunbar) proposed [*Package Manager Extensible Build Tools*](https://github.com/aciidb0mb3r/swift-evolution/blob/extensible-tool/proposals/NNNN-package-manager-extensible-tools.md). This proposal will allow community build tools ([Sourcery](https://github.com/krzysztofzablocki/Sourcery), [SwiftLint](https://github.com/realm/SwiftLint), [Jazzy](https://github.com/realm/jazzy), etc) to integrate with the *Swift Package Manager*.
+
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: You can remove this empty line; one should suffice.

File: _drafts/2018-03-23-issue-105.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+ 
+[SE-0192](https://github.com/apple/swift-evolution/blob/master/proposals/0192-non-exhaustive-enums.md) *Non-Exhaustive Enums* is in [re-review](https://forums.swift.org/t/se-0192-non-exhaustive-enums/7291).
+ 
+> The second review of the review of SE 0192 — Non-Exhaustive Enums begins now and runs through March 23, 2018.
+>
+>  The second review is follow up to the [original review](https://forums.swift.org/t/se-0192-non-exhaustive-enums/7291) based on feedback on the thread and from the Core Team.
+
+### Swift Forums
+
+[Ankit Aggarwal](https://github.com/aciidb0mb3r) & [Daniel Dunbar](https://github.com/ddunbar) proposed [*Package Manager Extensible Build Tools*](https://github.com/aciidb0mb3r/swift-evolution/blob/extensible-tool/proposals/NNNN-package-manager-extensible-tools.md). This proposal will allow community build tools ([Sourcery](https://github.com/krzysztofzablocki/Sourcery), [SwiftLint](https://github.com/realm/SwiftLint), [Jazzy](https://github.com/realm/jazzy), etc) to integrate with the *Swift Package Manager*.
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: Nit: `etcetera` over `etc`.

Comment (BasThomas) [reply _drafts/2018-03-23-issue-105.md]: Feel free to remove the italics for Swift Package Manager as well.

File: _drafts/2018-03-23-issue-105.md [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+
+
+> This is a proposal for adding package manager support for extensible build tools, i.e. executing tools at build time which were themselves produced by some other package, and whose behavior the package manager does not understood directly, but rather interacts through a well-defined, Swift, protocol.
+>  
+>  We expect this behavior to greatly enhance our capacity for building complex software packages.
+
+[Graydon Hoare](https://github.com/graydon) wrote about a new compilation mode called “batch” or “multiple file” mode.
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: Lets link to the thread on the forums here too.

File: _drafts/2018-03-23-issue-105.md (hunk 12 above)
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: We should link to the thread on the forums.

File: _drafts/2018-03-23-issue-105.md [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+
+> This new mode takes the set of files that would normally be (incrementally) compiled with individual subprocesses, and assigns them dynamically to multiple batches (one per core), compiling each batch with a single process. This means that incremental mode is still operative, and compilation makes use of multiple cores, but the benefits of whole-module mode (fewer processes, less redundant work) also apply. 
+  
+If you’re feeling adventurous and want to give it a try, the instructions are present on [Swift Forums](https://forums.swift.org/t/compilation-speed-help-test-batch-mode/10964)
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: `on the*`

File: _drafts/2018-03-23-issue-105.md [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+
+[Graydon Hoare](https://github.com/graydon) wrote about a new compilation mode called “batch” or “multiple file” mode.
+
+> This new mode takes the set of files that would normally be (incrementally) compiled with individual subprocesses, and assigns them dynamically to multiple batches (one per core), compiling each batch with a single process. This means that incremental mode is still operative, and compilation makes use of multiple cores, but the benefits of whole-module mode (fewer processes, less redundant work) also apply. 
+  
+If you’re feeling adventurous and want to give it a try, the instructions are present on [Swift Forums](https://forums.swift.org/t/compilation-speed-help-test-batch-mode/10964)  
+
+### Finally
+
+[Look](https://twitter.com/chriseidhof) at [Chris Eidhof](https://twitter.com/chriseidhof) explaining MVC (Massive View Controller) 😀😆🤣.
```
Comment (BasThomas) [inline _drafts/2018-03-23-issue-105.md]: Nit: omit the `(Massive View Controller)`.

File: _drafts/2018-03-23-issue-105.md (hunk 1 above)
Comment (tapthaker) [reply _drafts/2018-03-23-issue-105.md]: I am trying to find something except for WWDC, so lets see. Will delete the News & Community if I don't find anything interesting.

File: _drafts/2018-03-23-issue-105.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+[Graydon Hoare](https://github.com/graydon) wrote about a new compilation mode called “batch” or “multiple file” mode.
+
+> This new mode takes the set of files that would normally be (incrementally) compiled with individual subprocesses, and assigns them dynamically to multiple batches (one per core), compiling each batch with a single process. This means that incremental mode is still operative, and compilation makes use of multiple cores, but the benefits of whole-module mode (fewer processes, less redundant work) also apply. 
+  
+If you’re feeling adventurous and want to give it a try, the instructions are present on the [Swift Forums](https://forums.swift.org/t/compilation-speed-help-test-batch-mode/10964)  
+
+### Finally
+
+[Look](https://twitter.com/chriseidhof) at [Chris Eidhof](https://twitter.com/chriseidhof) explaining MVC. 😀😆🤣
```
Comment (jessesquires) [inline _drafts/2018-03-23-issue-105.md]: are these links wrong?

Comment (BasThomas) [reply _drafts/2018-03-23-issue-105.md]: You beat me 😀

**Comment by jessesquires** [comment]
Hey @tapthaker 

We really need to improve here. We're missing a lot of content from #372, and today is publish day.

I think we really need to focus on getting the first half of a draft finished during the first week of the writing cycle. It will help reduce a lot of the stress.

**Comment by tapthaker** [comment]
I understand. Give me a few hours to get last few content out. Will ensure that the last minute tension doesn’t happen again.

**Comment by BasThomas** [comment]
I'll add some content in the meantime @tapthaker. :)

**Comment by tapthaker** [comment]
@BasThomas : I am starting to work on it as well. How do we distribute ?

**Comment by BasThomas** [comment]
Ah sorry - I just pushed some changes. Have a look @tapthaker.

**Comment by tapthaker** [comment]
@BasThomas : Looks good to me. You are fast !!

**Comment by BasThomas** [comment]
Thanks @tapthaker!

**Comment by tapthaker** [comment]
No, thank you @BasThomas & @jessesquires for coming in at the last moment and pushing fixes !!

**Comment by jessesquires** [comment]
Thanks all! 💯

**Pull request merged**